src/phantom_token.c

```diff
@@ -0,0 +1,525 @@
+
+static ngx_int_t post_configuration(ngx_conf_t *config)
+{
+    ngx_http_core_main_conf_t *main_config = ngx_http_conf_get_module_main_conf(config, ngx_http_core_module);
+    ngx_http_handler_pt *h = ngx_array_push(&main_config->phases[NGX_HTTP_ACCESS_PHASE].handlers);
+
+    if (h == NULL)
+    {
+        return NGX_ERROR;
+    }
+
+    *h = handler;
+
+    return NGX_OK;
+}
+
+static void *create_location_configuration(ngx_conf_t *config)
+{
+    phantom_token_configuration_t *location_config = ngx_pcalloc(config->pool, sizeof(phantom_token_configuration_t));
+
+    if (location_config == NULL)
+    {
+        return NGX_CONF_ERROR;
+    }
+
+    location_config->enable = NGX_CONF_UNSET_UINT;
+    location_config->scopes = NGX_CONF_UNSET_PTR;
+
+    return location_config;
+}
+
+static char *merge_location_configuration(ngx_conf_t *main_config, void *parent, void *child)
+{
+    phantom_token_configuration_t *parent_config = parent, *child_config = child;
+
+    ngx_conf_merge_off_value(child_config->enable, parent_config->enable, 0)
+    ngx_conf_merge_str_value(child_config->introspection_endpoint, parent_config->introspection_endpoint, "")
+    ngx_conf_merge_str_value(child_config->realm, parent_config->realm, "api")
+    ngx_conf_merge_ptr_value(child_config->scopes, parent_config->scopes, NULL)
+    ngx_conf_merge_str_value(child_config->space_separated_scopes, parent_config->space_separated_scopes, "")
+
+    if (child_config->scopes != NULL && child_config->space_separated_scopes.len == 0)
+    {
+        // Flatten scopes into a space-separated list
+        ngx_str_t *scope = child_config->scopes->elts;
+        size_t space_separated_scopes_data_size = child_config->scopes->nelts;
+        ngx_uint_t i;
+
+        for (i = 0; i < child_config->scopes->nelts; i++)
+        {
+            space_separated_scopes_data_size += scope[i].len;
+        }
+
+        u_char *space_separated_scopes_data = ngx_pcalloc(main_config->pool, space_separated_scopes_data_size);
+
+        if (space_separated_scopes_data == NULL)
+        {
+            return NGX_CONF_ERROR;
+        }
+
+        u_char *p = space_separated_scopes_data;
+
+        for (i = 0; i < child_config->scopes->nelts; i++)
+        {
+            p = ngx_cpymem(p, scope[i].data, scope[i].len);
+            *p = ' ';
+            p++;
+        }
+
+        *(p - 1) = '\0';
+
+        child_config->space_separated_scopes.data = space_separated_scopes_data;
+        child_config->space_separated_scopes.len = ngx_strlen(space_separated_scopes_data);
+
+        assert(child_config->space_separated_scopes.len <= space_separated_scopes_data_size);
+    }
+
+    return NGX_CONF_OK;
+}
+
+/**
+ * The main handler logic
+ */
+ngx_int_t handler(ngx_http_request_t *request)
+{
+    phantom_token_configuration_t *module_location_config = ngx_http_get_module_loc_conf(
+            request,
+            ngx_curity_http_phantom_token_module);
+
+    // Return OK if the module is not active
```

Comment:
Is the comment or the response wrong? we say return OK but we return NGX_DECLINED

Reply:
Ah - fixed comment